### DISCUSSION TITLE
How to deal with data migration?

### DISCUSSION THREAD

**Discussion by Hugoo** [prisma/prisma#6031] [Q&A] [answered]
Hi,

I could not find any information in the docs regarding "data migration".
Is there a way to attach a script along a migration when data need to be updated?

Exactly as it is described here:
https://docs.djangoproject.com/en/3.1/topics/migrations/#data-migrations

Let's say I had a `firstname` and `lastname` columns and I want to perform a migration that removes these columns and creates a new one `name` which is firstname + lastname ?

How would I attach my custom migration script with prisma migrations?

Thanks

**Comment by ryands17** [comment, marked as answer]
@Hugoo 👋 

One way to go about this would be [creating the required field](https://www.prisma.io/docs/concepts/components/prisma-migrate) and running Prisma's seed command. Considering a schema like this:

```prisma
model User {
  id       Int    @id @default(autoincrement())
  firstName String
  lastName String
}
```

You can add a new column by editing the above schema:

```prisma
model User {
  id        Int     @id @default(autoincrement())
  name  String?
  firstName String
  lastName  String
}
```

The the following commands:

```
prisma migrate dev --create-only --preview-feature
prisma generate
```

This will generate the migrations files and now you can edit the seed file that can be created as shown [here](https://www.prisma.io/docs/guides/prisma-guides/seed-database). In this script, you can append the `firstName` and `lastName` to the `name` field.

Finally apply the migration and seed file:

```
prisma migrate dev --preview-feature
prisma db seed --preview-feature
```

This is how you can deal with data migration. Let me know if you have any other queries :)

**Reply by Hugoo** [reply]
Thanks for the update, I'll give it a try !

**Reply by bryanmig** [reply]
@ryands17 As I understand it, there is only a single `seed.ts` file and no way to run code with a specific migration.  Is that right?

In this sense, the `seed.ts` isn't really a great solution for migrations like this. Is there any planned feature that is more like the one that is linked to above?  This is one of my big concerns.

**Reply by ryands17** [reply]
@bryanmig 👋 
Yes this currently planned in the roadmap [here](https://www.notion.so/prismaio/Better-abstraction-for-manipulating-data-alongside-schema-migrations-14656f377f3b4531841ddc6b917863c2) to allow better data migrations, but currently a separate script or using a seed script is the only workaround.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
